### PR TITLE
Graph API updates

### PR DESCRIPTION
**Pull request by cquiroz**
This PR adds several changes to the graph api:

* Support to request only one axis of dat
* Support to request a description of the axes (min,max,etc...))
* Support to truncate the amount of significant figures. This to reduces the payload and hide the approximation errors produced by ITC

### CODE REVIEW

File: modules/itc/src/main/scala/lucuma/itc/ItcChart.scala [hunk 1]
```diff
@@ -4,31 +4,43 @@
...
 
-enum SeriesDataType:
-  case SignalData, BackgroundData, SingleS2NData, FinalS2NData, PixSigData, PixBackData
+enum SeriesDataType(val tag: String):
+  case SignalData     extends SeriesDataType("signal_data")
+  case BackgroundData extends SeriesDataType("background_data")
+  case SingleS2NData  extends SeriesDataType("single_s2_ndata")
+  case FinalS2NData   extends SeriesDataType("final_s2_ndata")
+  case PixSigData     extends SeriesDataType("pix_sig_data")
+  case PixBackData    extends SeriesDataType("pick_back_data")
```
Comment (rpiaggio) [inline modules/itc/src/main/scala/lucuma/itc/ItcChart.scala:30]: Typo in the tag?

File: modules/itc/src/main/scala/lucuma/itc/SignificantFigures.scala [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+package lucuma.itc
+
+import eu.timepit.refined.types.numeric.PosInt
+import io.circe._
+import io.circe.refined._
+
+import scala.math._
+
+case class SignificantFigures(xAxis: Option[PosInt], yAxis: Option[PosInt])
```
Comment (rpiaggio) [inline modules/itc/src/main/scala/lucuma/itc/SignificantFigures.scala:12]: final?

Comment (cquiroz) [reply modules/itc/src/main/scala/lucuma/itc/SignificantFigures.scala:12]: I learn we don't really need `final` anymore
https://medium.com/scala-3/opaque-type-aliases-and-open-classes-13076a6c07e4

File: modules/itc/src/main/scala/lucuma/itc/ItcChart.scala (hunk 1 above)
Comment (cquiroz) [reply modules/itc/src/main/scala/lucuma/itc/ItcChart.scala:30]: Great catch

**Review by rpiaggio — APPROVED** [review]
LGTM

**Pull request merged**